tests/unit/test_experiments/test_simulation_with_experiment.py

```diff
@@ -261,77 +261,84 @@ def test_solve_with_sensitivities_and_experiment(self):
         experiment_2step = pybamm.Experiment(
             [
                 (
-                    "Discharge at C/20 for 1 hour",
-                    "Charge at 1 A until 4.1 V",
-                    "Hold at 4.1 V until C/2",
-                    "Discharge at 2 W for 30 min",
-                    "Discharge at 2 W for 30 min",  # repeat to cover this case (changes initialisation)
+                    "Discharge at C/20 for 2 min",
+                    "Charge at 1 A for 1 min",
+                    "Hold at 4.1 V for 1 min",
+                    "Discharge at 2 W for 1 min",
+                    "Discharge at 2 W for 1 min",  # repeat to cover this case (changes initialisation)
                 ),
             ]
             * 2,
         )
 
         solutions = []
-        for solver in [
-            pybamm.CasadiSolver(),
-            pybamm.IDAKLUSolver(),
-            pybamm.ScipySolver(),
-        ]:
-            for calculate_sensitivities in [False, True]:
-                model = pybamm.lithium_ion.SPM()
-                param = model.default_parameter_values
-                input_param_name = "Negative electrode active material volume fraction"
-                input_param_value = param[input_param_name]
-                param.update({input_param_name: "[input]"})
-                sim = pybamm.Simulation(
-                    model,
-                    experiment=experiment_2step,
-                    solver=solver,
-                    parameter_values=param,
-                )
-                solution = sim.solve(
-                    inputs={input_param_name: input_param_value},
-                    calculate_sensitivities=calculate_sensitivities,
-                )
-                solutions.append(solution)
+        input_param_name = "Negative electrode active material volume fraction"
+        solver = pybamm.IDAKLUSolver(atol=1e-8, rtol=1e-8)
+        for calculate_sensitivities in [False, True]:
+            model = pybamm.lithium_ion.SPM()
```

Comment:
Can we test this with a DAE model like the `DFN` to make sure DAE initialization is handled appropriately?
```suggestion
            model = pybamm.lithium_ion.DFN()
```

Reply:
yup, done this now